posts/2020/08/the-ibis-backends.md

```diff
@@ -0,0 +1,359 @@
+For the past 3 years, Quansight, in partnership with OmniSci, added the `ibis.omniscidb`
+string generating backend. Further, our responsibilities have expanded
+to support `ibis` as community maintainers through Quansight Labs. 
+This collaboration introduced geospatial functionality to `ibis` for several backends.
+
+### Currently, there is an ongoing effort to add sqlserver backends
+
+There are on going efforts to introduce [SQL Server][sql-server] support.  
+What other backends would you like to see for `ibis`?
+Maybe our community could benefit from a `dask` backend or `altair` backend?
+
+## `ibis` direct execution backends
+
+`ibis` direct execution backends like `pandas` and `hdf5` operate on conventional in-memory python objects.
+`pandas` is the gold standard for structured data in python, and inspires the API for `ibis`.
+
+
+```python
+    pd = ibis.pandas.connect({'A': pandas.util.testing.makeDataFrame()})
+```
+
+`pd` is an `ibis` backend based off `pandibis direct execution backends like pandas and hdf5 operate on conventional in-memory python objects. pandas is the gold standard for structured data in python, and inspires the API for ibis.
+
+as.DataFrame` objects.
+
+```python
+    expression = pd.table('A').head()
+```
+
+`expression` is an `ibis` query, that has `expression.compile` and `expression.execute` methods.
+We'll recognize the __execute__ method when we return `pandas.DataFrame`s from `ibis` expression.
+The __compile__ method does not trigger any computation, rather it constructs an intermediate form
+that is interpretted by a backend.
+
+```python
+    >>> assert isinstance(expression.compile(), ibis.expr.types.TableExpr)
+```
+
+In the case of direction execution backends, the `expression` compiles to an the original `ibis` 
+expression.  And the computation is carried out based on a set of recipes defined in `ibis`.
+
+In general, we would typically do this work directly in `pandas`, however this work is
+practical in mocking tests for expressions independent of backends.
+
+> Learn more about the [HDF5 direct execution backend in the `ibis` tests][test-hdf5].
+
+## `ibis` expression generating backends.
+
+```python
+    db = ibis.sqlite.connect('lahmansbaseballdb.sqlite')
+    expression = db.table('halloffame').head()
+```
+
+Expression generating backends operate on [SQL] databases that interoperate with `sqlalchemy`.
+
+```python
+    >>> assert isinstance(expression.compile(), sqlalchemy.sql.Select)
+```
+
+In the case of expression generating backends, the intermediate representation is a `sqlalchemy` object.
+`sqlalchemy` is _The Database Toolkit for Python_, and `ibis` leverages it compatability
+with traditional [SQL] databases.
+    
+
+
+## `ibis` string generating backends.
+
+```bash
+pip install --upgrade ibis-framework[omniscidb]
+# ORx
+conda install -c conda-forge ibis-framework # install all the backends!
```

Comment:
this comment is confusing, it tells the reader that `pip install` does not get you all the backends, but which ones are missing? better be explicit, and move this into regular text